### PR TITLE
Add resource type to execution completed

### PR DESCRIPTION
**Pull request by nelsonkopliku**
Adds a `ResourceType` definition that allows to identify the resource for which a check execution is being run.

Currently `host` and `cluster` are made available. As we add support for more checks scopes (ie: sap systems) we'll add more definitions.

Added the resource type to `ExecutionCompleted` by now. As we proceed with the implementation of host checks we will update other needed protos, for instance `ExecutionRequested`.

### CODE REVIEW

File: protobuf/resource_type.proto [hunk 1]
```diff
@@ -0,0 +1,10 @@
+syntax = "proto3";
+
+package Trento.Checks.V1;
+option go_package = "/;events";
+
+enum ResourceType {
+    UNKNOWN = 0;
```
Comment (CDimonaco) [inline protobuf/resource_type.proto:7]: My bad today, I said unknown but if we want to stick to the convention, use unspecified https://protobuf.dev/programming-guides/dos-donts/#do-include-an-unspecified-value-in-an-enum

**Review by CDimonaco — CHANGES_REQUESTED** [review]

**Review by arbulu89 — DISMISSED** [review]
Perfect

> Added the resource type to ExecutionCompleted by now. As we proceed with the implementation of host checks we will update other needed protos, for instance ExecutionRequested.

Alright!

**Review by arbulu89 — COMMENTED** [review]
Hey @nelsonkopliku ,
With the new `0.11.0` version, checking the changelog, I have noticed that they have deprecated the `new/new!` functions. https://github.com/elixir-protobuf/protobuf/blob/main/CHANGELOG.md#deprecations
I think this will be an issue, as we use them in our mapping.
Could you have a look?
Maybe worth to defer to upgrade of version if that's the case...

**Review by arbulu89 — APPROVED** [review]
@nelsonkopliku Ok!
Please track the debt
(I wouldn't add these changes in the work you will do. I would create individual PRs to fix this so we have the PR really visible and documented)

**Review by dottorblaster — APPROVED** [review]
@arbulu89 you're right, let's tackle this as soon as possible when we touch Wanda and the Web. I don't like too much the new `.struct()` but it is what it is :smile:

**Review by CDimonaco — APPROVED** [review]

**Pull request merged**